Work Order | test_valuation_rate_missing_on_make_stock_entry |     self.assertRaises(frappe.ValidationError, stock_entry.save)
AssertionError: ValidationError not raised by save

### Issue Details:
    self.assertRaises(frappe.ValidationError, stock_entry.save)
AssertionError: ValidationError not raised by save

### Traceback:
``` 
FAIL: test_valuation_rate_missing_on_make_stock_entry (erpnext.manufacturing.doctype.work_order.test_work_order.TestWorkOrder)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/manufacturing/doctype/work_order/test_work_order.py", line 909, in test_valuation_rate_missing_on_make_stock_entry
    self.assertRaises(frappe.ValidationError, stock_entry.save)
AssertionError: ValidationError not raised by save
```

--- a/erpnext/stock/stock_ledger.py
+++ b/erpnext/stock/stock_ledger.py
@@ -1838,8 +1838,8 @@ def get_valuation_rate(
 			)
 
 	if (
-		not allow_zero_rate
-		and not valuation_rate
+		(valuation_rate is None or flt(valuation_rate) == 0)
+		and not cint(allow_zero_rate)
 		and raise_error_if_no_rate
 		and cint(erpnext.is_perpetual_inventory_enabled(company))
 	):

